base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```

Comment:
`nameable` should be accessible from outside the class by class methods. Kindly, add  `attr_accessor` to `nameable`.

Reply:
@muneebulrehman Thank you for the review!! I have one small question, do I make it accessible inside the base decorator class?